llvm/lib/CodeGen/BasicBlockSectionsProfileReader.cpp

```diff
@@ -213,10 +213,6 @@ Error BasicBlockSectionsProfileReader::ReadV1Profile() {
               Twine("duplicate basic block id found '") + BasicBlockIDStr +
               "'");
 
-        if (!BasicBlockID->BaseID && CurrentPosition)
-          return createProfileParseError(
-              "entry BB (0) does not begin a cluster.");
-
```

Comment:
I see, so you are removing the error when "0" does not begin a cluster and forcing the entry block to be the beginning without any message.  I am fine with this change but we should add a note somewhere about this enforcement.